packages/server/tests/AuthenticationSignUp.test.ts

```diff
@@ -70,12 +93,86 @@ test.group('Authentication sign up tests group', (group) => {
         assert.equal(getMyProfileParsedBody.userID, userID);
     });
 
+    test('It should send an email verification email', async (assert) => {
+        const email = internet.email();
+        const nickname = internet.userName();
+        const password = generateStrongPassword();
+
+        const user = await User.create({
+            email,
+            nickname,
+            password,
+        });
+
+        Mail.trap((message) => {
+            assert.deepEqual(message.to, [
+                {
+                    address: email,
+                },
+            ]);
+
+            assert.deepEqual(message.from, {
+                address: 'no-reply@adonisenprovence.com',
+            });
+
+            assert.equal(
+                message.subject,
+                `Welcome ${nickname}, please verify your email !`,
+            );
+        });
+
+        const emailVerification = new EmailVerification(user, 'token1234');
+        await emailVerification.send();
+
+        //Mail.trap will be expected without any more required actions such as waitFor
+    });
+
+    test.failing(
+        'It should fail expecting no email to be sent',
+        async (assert) => {
+            const email = 'cinem69586@f1xm.com'; //internet.email();
+            const nickname = internet.userName();
+            const password = generateStrongPassword();
+
+            const user = await User.create({
+                email,
+                nickname,
+                password,
+            });
+
+            Mail.trap(() => {
+                const onSignUpFailEmailShouldNotBeSent = true;
+                assert.isFalse(onSignUpFailEmailShouldNotBeSent);
+            });
+
+            const emailVerification = new EmailVerification(user, 'token1234');
+            await emailVerification.send();
+        },
+    );
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Why should this test fail? What does make it different from the very above test?

Reply:
Here I'm testing that the following assertion:
```ts
        Mail.trap(() => {
                const expectingNoEmailToBeReceived = true;
                assert.isFalse(expectingNoEmailToBeReceived);
            });
```
Is failing when an email is received inside the Mail.trap. In this way I can be sure that when I'm using it inside test that should not receive email it would fail after receiving one

Reply:
Anyway ended up refactoring everything using sinon.spy